app/src/main/kotlin/org/cru/godtools/ui/languages/LanguageName.kt

```diff
@@ -41,24 +41,32 @@ internal fun LanguageName(language: Language, modifier: Modifier = Modifier) {
 
     LanguageName(
         displayName = remember(context, language, appLanguage) { language.getDisplayName(context, appLanguage) },
-        nativeName = remember(context, language) { language.getDisplayName(context, language.code) },
+        nativeName = if (language.isForcedName) {
+            null
+        } else {
+            remember(context, language) { language.getDisplayName(context, language.code) }
+        },
         modifier = modifier,
     )
 }
 
 private const val LANGUAGE_NAME_GAP = "[gap]"
 
 @Composable
-private fun LanguageName(displayName: String, nativeName: String, modifier: Modifier = Modifier) {
+private fun LanguageName(displayName: String, modifier: Modifier = Modifier, nativeName: String? = null) {
     val color = LocalTextStyle.current.color.takeOrElse { LocalContentColor.current }
     val displayNameColor = color.let { it.copy(alpha = it.alpha * 0.60f) }
 
     Text(
         remember(displayName, nativeName, color, displayNameColor) {
             buildAnnotatedString {
-                withStyle(SpanStyle(color = color)) { append(nativeName) }
-                appendInlineContent(LANGUAGE_NAME_GAP, " ")
-                withStyle(SpanStyle(color = displayNameColor)) { append(displayName) }
+                if (nativeName == null) {
+                    withStyle(SpanStyle(color = color)) { append(displayName) }
+                } else {
+                    withStyle(SpanStyle(color = color)) { append(nativeName) }
+                    appendInlineContent(LANGUAGE_NAME_GAP, " ")
+                    withStyle(SpanStyle(color = displayNameColor)) { append(displayName) }
+                }
```

Comment:
this can be simplified if you separate the concept of display and native name from this Composable and just use a generic `name` and `secondaryName`. it will require swapping the "native" and "display" names where you call the composable.

then the logic could look like:
```
buildAnnotatedString {
    withStyle(SpanStyle(color = color)) { append(name) }
    if (secondaryName != null) {
        appendInlineContent(LANGUAGE_NAME_GAP, " ")
        withStyle(SpanStyle(color = secondaryNameColor)) { append(secondaryName) }
    }
}
```

this would yield a more straightforward composable that is easier to understand